LAKMobile/src/api/consumer.tsx

```diff
@@ -294,3 +300,59 @@ export const createNewUser = async (formData: CreateUserForm): Promise<string |
     return null;
   }
 };
+
+/**
+ *
+ * @param userId ID of user applying to job
+ * @param jobId ID of job that user is applying to
+ * @returns
+ */
+export const applyJob = async (
+  userId: string,
+  jobId: string
+): Promise<{ userId: string; jobId: string } | null> => {
+  try {
+    const url = `${GET_JOBS}/${jobId}/apply?` + new URLSearchParams({ user: userId });
+    const response = await fetch(url, {
+      method: 'PATCH',
+    });
+    const data = await response.json();
+    return { userId: data.userId, jobId: data.jobId };
+  } catch {
+    return null;
+  }
+};
+
+// Not Started | Applied | Accepted | Denied | Finished
+export const getJobApplicantStatus = (
+  job: JobOwnerView,
+  userId: string
+): 'Not Started' | 'Applied' | 'Accepted' | 'Denied' | 'Finished' => {
+  const applicants = job.applicants || [];
+  if (applicants.find((applicant) => applicant.userId === userId)) {
+    const isAssignedDriver = job.assignedDriverId === userId;
+    switch (job.status) {
+      case 'CREATED':
+        return 'Applied';
+      case 'ASSIGNED':
+        return isAssignedDriver ? 'Accepted' : 'Denied';
+      case 'COMPLETED':
+        return isAssignedDriver ? 'Finished' : 'Denied';
+    }
+  } else {
+    return job.status === 'CREATED' ? 'Not Started' : 'Finished';
+  }
+};
+
+export const completeJob = async (userId: string, jobId: string): Promise<string | null> => {
+  try {
+    const url = `${GET_JOBS}/${jobId}/complete?` + new URLSearchParams({ user: userId });
+    const response = await fetch(url, {
+      method: 'PATCH',
+    });
+    const data = await response.json();
+    return data.jobId;
```

Comment:
For completeJob I was wondering why we only return jobId and not userId also?

Reply:
Also how do you know what to return? Why is it necessary to return anything at all?

Reply:
I'm basically following what was returned from the backend here and returning it on the frontend; it makes sense to return jobId and not userId since it's guaranteed that the driver assigned to the job will always be marking the job as complete. It's not completely necessary to return anything here, but it can be helpful to just have everything available to the developer just in case.